### PR TITLE
aichat sidebar

### PR DESCRIPTION
**Pull request by adred**

### CODE REVIEW

File: src/app/sidebar/aichat.tsx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        }).catch((_) => {});
+    }
+
+    getLinePos(elem: any): { numLines: number; linePos: number } {
+        const numLines = elem.value.split("\n").length;
+        const linePos = elem.value.substr(0, elem.selectionStart).split("\n").length;
+        return { numLines, linePos };
+    }
+
+    @mobx.action.bound
```
Comment (esimkowitz) [inline src/app/sidebar/aichat.tsx:196]: you'll need to add a constructor that calls `mobx.makeObservable(this)` in order for these attributes to work

File: src/app/sidebar/aichat.tsx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                        {this.renderChatMessage(chitem)}
+                    </For>
+                </div>
+            </OverlayScrollbarsComponent>
+        );
+    }
+}
+
+@mobxReact.observer
+class AIChat extends React.Component<{}, {}> {
```
Comment (esimkowitz) [inline src/app/sidebar/aichat.tsx:159]: Can we not reuse more from the aichat auxview or maybe make that a component somewhere else that both views can utilize? Same for the keybindings

**Comment by sawka** [comment]
red, you're going to have to resolve the conflicts here -- including migrate to use clsx (instead of classnames) and the new "yarn modern".

**Comment by sawka** [comment]
closing, since I think this is the old PR.  I don't want to get confused.

**Pull request closed without merging**